pg_shard.c

```diff
@@ -579,7 +579,7 @@ ExtractFirstDistributedTableId(Query *query)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
 
-		if (IsDistributedTable(rangeTableEntry->relid))
+		if (rangeTableEntry->rtekind == RTE_RELATION && IsDistributedTable(rangeTableEntry->relid))
```

Comment:
`0` is the same as `InvalidOid`. Right now I think the code "just works" and vacuously returns `false`, but we pass the `oid` argument to `get_rel_namespace`, so we're now making an assumption that it will not barf on `InvalidOid` input.

A bigger issue is: can we trust that it's actually zero for non-relations? What if somewhere in the PostgreSQL code `palloc` is used instead of `palloc0`? Then we've got a bogus relation identifier being passed to this function, which could have undefined behavior.

After reasoning this out here, I'm thinking either `IsDistributedTable` should just be more explicit in its handling of `InvalidOid`: it should error out if we aren't going to handle it, or return early if we will.

Since I do like the idea of allowing it to simply handle `InvalidOid`, I'll just have it short-circuit if it sees that argument. Sound OK? (done)